controllers/apps/suite_test.go

```diff
@@ -49,53 +48,55 @@ func testClient() client.Client { return k8sClient }
 
 func TestAPIs(t *testing.T) {
 	RegisterFailHandler(Fail)
-
-	RunSpecsWithDefaultAndCustomReporters(t,
-		"Controller Suite",
-		[]Reporter{printer.NewlineReporter{}})
+	RunSpecs(t, "Controller Suite")
 }
 
 var _ = BeforeSuite(func() {
-	// Set controller-runtime output to GinkgoWriter
-	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
-
-	By("bootstrapping test environment")
-	testEnv = &envtest.Environment{
-		CRDDirectoryPaths:     []string{filepath.Join("..", "..", "config", "crd", "bases")},
-		ErrorIfCRDPathMissing: true,
-	}
+	done := make(chan interface{})
+	go func() {
+		// Set controller-runtime output to GinkgoWriter
+		logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
 
-	var err error
-	cfg, err = testEnv.Start()
-	Expect(err).ToNot(HaveOccurred())
-	Expect(cfg).ToNot(BeNil())
+		By("bootstrapping test environment")
+		testEnv = &envtest.Environment{
+			CRDDirectoryPaths:     []string{filepath.Join("..", "..", "config", "crd", "bases")},
+			ErrorIfCRDPathMissing: true,
+		}
 
-	err = appsv1alpha1.AddToScheme(scheme.Scheme)
-	Expect(err).NotTo(HaveOccurred())
+		var err error
+		cfg, err = testEnv.Start()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(cfg).ToNot(BeNil())
 
-	// +kubebuilder:scaffold:scheme
+		err = appsv1alpha1.AddToScheme(scheme.Scheme)
+		Expect(err).NotTo(HaveOccurred())
 
-	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
-	Expect(err).NotTo(HaveOccurred())
-	Expect(k8sClient).NotTo(BeNil())
+		// +kubebuilder:scaffold:scheme
 
-	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
-		Scheme: scheme.Scheme,
-	})
-	Expect(err).ToNot(HaveOccurred())
-	err = (&APIcastReconciler{
-		BaseControllerReconciler: reconcilers.NewBaseControllerReconciler(mgr.GetClient(), mgr.GetAPIReader(), mgr.GetScheme()),
-		Log:                      ctrl.Log.WithName("controllers").WithName("APIcast"),
-	}).SetupWithManager(mgr)
-	Expect(err).ToNot(HaveOccurred())
+		k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(k8sClient).NotTo(BeNil())
 
-	go func() {
-		defer GinkgoRecover()
-		err = mgr.Start(ctrl.SetupSignalHandler())
+		mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+			Scheme: scheme.Scheme,
+		})
+		Expect(err).ToNot(HaveOccurred())
+		err = (&APIcastReconciler{
+			BaseControllerReconciler: reconcilers.NewBaseControllerReconciler(mgr.GetClient(), mgr.GetAPIReader(), mgr.GetScheme()),
+			Log:                      ctrl.Log.WithName("controllers").WithName("APIcast"),
+		}).SetupWithManager(mgr)
 		Expect(err).ToNot(HaveOccurred())
-	}()
 
-}, 60)
+		go func() {
```

Comment:
another goroutine? really needed?

Reply:
The bump to ginkgo v2 changed the function signature and we can no longer pass in the `60` timeout using the old method. For what it's worth, I've found the test to be much faster and better with teardown using the goroutine.

Reply:
What is the value provided? I might be missing something, but I do not think we need that sync mechanism you implemented with the channel. 

Check the `suite_test.go` in the [kuadrant operator](https://github.com/Kuadrant/kuadrant-operator/blob/main/controllers/suite_test.go). It's working well as far as I can tell.

Reply:
@eguzki The value provided is that we preserve the existing behavior of the `60` timeout like we had with ginkgo v1. I'm also confused by the kuadrant-operator example because it looks like it also has that [goroutine](https://github.com/Kuadrant/kuadrant-operator/blob/main/controllers/suite_test.go#L230-L234) unless we're referring to different code blocks.

We've been using this implementation with [RHOAM](https://github.com/integr8ly/integreatly-operator/blob/master/test/functional/suite_test.go#L79-L106) for awhile now and haven't had any issues with it.

Reply:
I am fine with this goroutine, it is well known. I was asking for the other one in the top and the sync mechanism to respect the 60 secs. 
```go
done := make(chan interface{})
go func() {
....
}
Eventually(done, 60).Should(BeClosed())
```

My understanding (corrrect me if I am wrong), is that this is enforcing the test setup (not the entire test) to happen within 60sec time period. The test setup being starting the test environment and instantiate the managers and controllers and finally, start the manager. That makes sense, when using the kubebuilder test environment, a kind of k8s fake control plane. However, we are using a real k8s control plane which is started before running the e2e tests. So, does it still makes sense? I am ok to keep that, not a big deal.

Reply:
Ah okay that makes sense thank you for the explanation. I fixed it and pushed the changes